src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -121,10 +121,10 @@ bool Exceptions::special_exception(JavaThread* thread, const char* file, int lin
                         "throwing pre-allocated exception: %s",
                         exc_value, message ? ": " : "", message ? message : "",
                         p2i(h_exception()), file, line, p2i(thread),
-                        Universe::vm_exception()->print_value_string());
+                        Universe::null_ptr_exception_instance()->print_value_string());
```

Comment:
I think this will look odd. Throwing `InternalError` may be more appropriate.

Reply:
Unfortunately, we don't pre-allocate an instance of InternalError.

    // We do not care what kind of exception we get for a thread which
    // is compiling.  We just install a dummy exception object
    thread->set_pending_exception(Universe::null_ptr_exception_instance(), file, line);

Reply:
I have to put all this code back to create, store for CDS and use InternalError.

Reply:
> Unfortunately, we don't pre-allocate an instance of InternalError.

But you just need to replaced the lines that you deleted for VirtualMachineError with InternalError. I think that would be cleaner than having to explain a weird behavior in comments.

Reply:
Seems like a waste, but I can put the lines back and make them InternalError.

Reply:
Ok, I made VirtualMachineError into InternalError for this, and restored the CDS code.